### ISSUE TITLE
error during point assignment in JBA 3.1   (same coordinate works with JBA 3.0.3)

### ISSUE THREAD

**Issue by SBludau** [FZJ-INM1-BDA/siibra-explorer#1468]
Coordinate 32mm, -28mm, -8mm  gives CA3 in JBA 3.0.3, which is correct. After a switch to JBA 3.1 the position seems to be correct as well, so the crosshair is located at the correct position, but the assignment produces an error.

**JBA 3.0.3 Output**
![image](https://github.com/user-attachments/assets/c52049d0-af04-4dff-a573-49a8dfa683d6)

**JBA 3.1 Output**
![image](https://github.com/user-attachments/assets/3f80a9cc-2cb9-412a-a468-d69ce8cf32f0)

@ Xiao: could you provide an estimation of the timeframe required to address this issue. The assignment is needed for an update to a publication that is currently in its final stages. (However, I want to emphasize that there is no immediate urgency, as the lead author is on vacation for 2 weeks ;-) )

**Comment by xgui3783** [comment]
TLDR: siibra-python is too memory hungry. next version of siibra should (hopefully) fix this issue.

As it stands, the point assignment will often fail (depending on if other users have used point assignment on other parcellations etc)

Please see below the same point assigned (hopefully correctly)

![Screenshot 2024-07-26 111049](https://github.com/user-attachments/assets/58861948-c2a3-499e-a6c0-bb1a76a88158)
[pointassignment.zip](https://github.com/user-attachments/files/16389498/pointassignment.zip)

**Comment by SBludau** [comment]
Now, the point assignment shows the value of the labeled map.

We recently used the point assignment with siibra_python for a publication and a dissertation. To be honest, we are a bit concerned that the values we are getting might not be "stable." Could you please confirm that the point assignment with siibra_python and JBA 3.1 is functioning correctly and will remain reproducible?

siibra explorer screenshot

<img width="917" alt="mappingvalue_JBv31_online" src="https://github.com/user-attachments/assets/2da208c6-83cd-46ee-9608-b2db4554ee8f">

siibra-python output
<img width="488" alt="mappingvalue_JBv31_script" src="https://github.com/user-attachments/assets/af23ff13-43f5-496c-9f87-51bef159c5f5">

**Comment by xgui3783** [comment]
Apologies. 

The performance of siibra-explorer/api is quite lacking when it comes to memory hungry tasks. We are in the process of remedying it in the coming version. 

I have for now freed up some resources, and point assignment can be performed again, but the result is somewhat worrying:

![Screenshot 2024-08-16 113521](https://github.com/user-attachments/assets/e36384b3-dfd6-4942-b346-463243833cf5)

full assigment:

[pointassignment (1).zip](https://github.com/user-attachments/files/16635622/pointassignment.1.zip)

@SBludau do you have the full output and/or do you know with version of siibra-python you were using?

@AhmetNSimsek would you know the discrepency? (siibra-api uses siibra-python v0.4a80)

**Comment by AhmetNSimsek** [comment]
This was a bug on version 1.0a08 of siibra-python and it was remedied by 1.0a09. I can provide more context @SBludau.

About assignment issues on explorer to 3.1, we have an idea. I am making a fix at the moment.

> Apologies.
> 
> The performance of siibra-explorer/api is quite lacking when it comes to memory hungry tasks. We are in the process of remedying it in the coming version.
> 
> ~I have for now freed up some resources, and point assignment can be performed again, but the result is somewhat worrying~:
> 
> (screenshot removed and full assignment removed)
> 
> ~@SBludau do you have the full output and/or do you know with version of siibra-python you were using?~
> 
> ~@AhmetNSimsek would you know the discrepency? (siibra-api uses siibra-python v0.4a80)~
> 
> edit: apologies for the previous comment. I misunderstood the assignment. The screenshot and previous attachment refer to the assignment of julich brain `3.0.3`, _not_ `3.1`.
> 
> at the moment, this operation cannot be carried out on the viewer, unfortunately. we will endeavor to make it available as soon as possible. (sparse index is not available, so it appears that all volumes are loaded in memory, which causes OOM, and crashes the runner.)
> 
> In a more powerful machine, the assignment can be done. But worryingly, with different ressult.
> 
> on 0.4a80: https://github.com/FZJ-INM1-BDA/iav-dep-test/actions/runs/10440230365/job/28909876225#step:5:540
> 
> ```
>                                region  map value
> 0                 Area PGp (IPL) left   0.001438
> 1                 Area PGa (IPL) left   0.543131
> 2                 Area PFm (IPL) left   0.383613
> 3                  Area PF (IPL) left   0.071818
> 4  Temporal-to-Parietal (GapMap) left   0.000193
> ```
> 
> on 1.0a8 and beyond https://github.com/FZJ-INM1-BDA/iav-dep-test/actions/runs/10440230365/job/28909876028#step:5:537
> 
> ```
>                                region  map value
> 0             Area TPJ (STG,SMG) left        0.0
> 1                 Area PGp (IPL) left   0.008563
> 2                 Area PGa (IPL) left   0.556611
> 3                 Area PFm (IPL) left   0.384303
> 4                  Area PF (IPL) left   0.008503
> 5  Temporal-to-Parietal (GapMap) left   0.000295
> ```
> 
> (this matches to the result that @SBludau quoted)
> 
> ping @AhmetNSimsek
> 
> probably still worthwhile investigating the discrependency

**Comment by xgui3783** [comment]
This issue was first addressed by https://siibra-explorer.readthedocs.io/en/latest/releases/v2.14.12/ . 

Point assignment in siibra-explorer where statistical maps exist should work. 

Please feel to reopen the ticket if the issue persist.
